addons/mail/models/mail_thread.py

```diff
@@ -2139,6 +2140,13 @@ def message_post(self, *,
                  )
             )
         partner_ids = list(partner_ids or [])
+        if channel_ids and not is_list_of(channel_ids, int):
+            raise ValueError(
+                _('Posting a message should receive channels as a list of IDs (received %(cids)s)',
+                  cids=repr(channel_ids),
+                 )
+            )
```

Comment:
This is a programming error message. This should ideally never happen, but I don't mind have it in case we introduce the issue and the message error is explicit enough to help fixing faster.

However, I would not make the string translatable. This is a waste of translator effort.